### ISSUE TITLE
Export to AVIF format: the `encoding speed` parameter should not be exposed in the GUI

### ISSUE THREAD

**Issue by victoryforce** [darktable-org/darktable#17921]
User control over the entire range of values ​​for this parameter will not improve the user experience and may, on the contrary, worsen it if the user is not aware of the impact of different values ​​on the result.

First of all, let's put the issue of image quality out of the question. Tests show that the results when changing the parameter value is visually indistinguishable. Yes, you will find a lot of theoretical speculation on the net about the possible impact, but practice shows that this is not true.

All values ​​up to and including 5 should definitely not be provided for users to choose from. These values ​​lead to huge increases in export time. At the same time, depending on the compression mode, we either do not even get a reduction in file size in return (in lossy mode), or we get a very insignificant one, which is not worth the huge amount of time spent (in lossless mode).

### Lossless compression

Since a picture is worth a thousand words, take a look at the graphs of the size of the created file and the time spent on its creation depending on the value of the `encoding speed` parameter. This is an example for one specific image, tests for other images show a very similar picture.

![avif - file sizes](https://github.com/user-attachments/assets/0b5feb47-59ea-42da-beb3-8b95f73eddd4)
![avif - encoding times](https://github.com/user-attachments/assets/9b5fafd1-99a1-4c4c-926d-86dcaae0d900)

What can we conclude here? Values ​​from 0 to 5 will waste an unreasonable amount of time for a negligible gain in file size. Values ​​9 and 10, although encoding very quickly, result in a noticeably larger file size. The middle ground is values ​​6, 7, and 8.

I would probably just hardcode 6, for the reasons that if the user wants to get a lossless export, then the priority is quality and therefore, probably, a slightly slower export would be expected...

### Lossy compression

Tests in this compression mode show that:

- At any values ​​of this parameter, the exported images are visually indistinguishable.
- At values ​​less than 6, the encoding time gradually increases to extremely uncomfortable values ​​without any gain (the quality remains unchanged, the file size only increases).
- As for the minimum size of the resulting file, it turned out that there is no single encoding speed that provides the minimum size. For some images, the smallest file was at speed 6, for others - at speeds 9 and 10 (which in my tests gave an identical byte-by-byte result), but at 7 and 8 the files are larger for some reason.

Conclusions:
- all values ​​less than 6 do not bring any benefit, they only make it worse, because the export time increases significantly
- it seems that in practice there are two best values, these are 6 and 10
- we cannot predict which of these values ​​will provide the smallest export file size, it is only obvious that at 10 we have the highest export speed

---

To sum up, the optimal solution is not to expose the parameter, but to set it to a hardcoded values for each compression mode. For lossless compression, the optimal value seems to be 6, for lossy it is best to choose 10.
